test/complex.jl

```diff
@@ -733,6 +733,20 @@ end
 @test log10(10+0im) === 1.0 + 0.0im
 @test log2(2+0im) === 1.0 + 0.0im
 
+# sign
+for T in (Float32, Float64)
+    z = Complex{T}(1)
+    @test typeof(sign(z)) == typeof(z)
+    z = Complex{T}(0)
+    @test typeof(sign(z)) == typeof(z)
+end
+for T in (Int32, Int64)
+    z = Complex{T}(1)
+    @test typeof(sign(z)) == typeof(float(z))
+    z = Complex{T}(0)
+    @test typeof(sign(z)) == typeof(float(z))
+end
+
```

Comment:
Do you think that a stability like this would be fine? I used
```javascript
sign(z::Complex) = z == 0? float(zero(z)) : z/abs(z)
```
Sure enough that's not the only way to achieve it, though.